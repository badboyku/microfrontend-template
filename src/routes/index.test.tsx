import * as React from 'react';
import { render } from '@testing-library/react';
import Routes from './index';
import type { Props } from './index';

const defaultProps = {};

const renderRoutes = (props: Props = defaultProps) => {
  return render(<Routes {...props} />);
};

describe('Component Routes', () => {
  describe('when called', () => {
    it('renders without crashing', () => {
      const { asFragment } = renderRoutes();

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
