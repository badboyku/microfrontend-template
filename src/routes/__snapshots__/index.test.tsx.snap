// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Routes when called renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Home
    </h1>
    <div
      class="hello-world"
    >
      Hello World!
    </div>
    <img
      alt="logo"
      class="logo"
      src="logo.svg"
    />
    <svg
      width="40"
    >
      logo.svg
    </svg>
    <h3>
      Env Vars
    </h3>
    <div>
      IS_DEV: 
      <span
        style="font-weight: bold;"
      >
        false
      </span>
    </div>
    <div>
      IS_PROD: 
      <span
        style="font-weight: bold;"
      >
        true
      </span>
    </div>
    <div>
      PUBLIC_URL: 
      <span
        style="font-weight: bold;"
      />
    </div>
    <h4>
      Custom React App Env Vars
    </h4>
    <div>
      Buildtime: REACT_APP_MY_ENVVAR: 
      <span
        style="font-weight: bold;"
      />
    </div>
    <div>
      Runtime: REACT_APP_MY_ENVVAR: 
      <span
        style="font-weight: bold;"
      />
    </div>
  </div>
</DocumentFragment>
`;
